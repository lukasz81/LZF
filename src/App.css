/*
  Off-white (Page Background) = #f4f4f4
  White (Widget Background) = #ffffff
  Dark Grey (Text, Secondary Buttons) = #474c55
  Green (Primary Buttons, Widget Title) = #6db420
  Yellow (Tertiary Buttons, Page Border) = #fec01e
*/
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
:root {
  --off-white: #f4f4f4;
  --white: #ffffff;
  --dark-grey: #474c55;
  --green: #6db420;
  --yellow: #fec01e; }

body {
  background-color: var(--off-white);
  margin: 0;
  padding: 0;
  font-family: 'Open Sans',  sans-serif; }

.display--flex {
  display: flex; }

.display--block {
  display: block; }

.align-self--center {
  align-self: center; }

.container {
  position: relative;
  max-width: 1200px;
  margin: 0 auto; }

a {
  text-decoration: none;
  color: var(--dark-grey); }

.text--decoration-none {
  text-decoration: none; }

.text--indent {
  text-indent: -10000000px; }

.text--transform-uppercase {
  text-transform: uppercase; }

.text--transform-capitalize {
  text-transform: capitalize; }

input .error {
  box-shadow: 0 0 20px var(--yellow); }
