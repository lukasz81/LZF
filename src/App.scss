/*
  Off-white (Page Background) = #f4f4f4
  White (Widget Background) = #ffffff
  Dark Grey (Text, Secondary Buttons) = #474c55
  Green (Primary Buttons, Widget Title) = #6db420
  Yellow (Tertiary Buttons, Page Border) = #fec01e
*/

@import url('https://fonts.googleapis.com/css?family=Open+Sans');

:root {
    --off-white: #f4f4f4;
    --white: #ffffff;
    --dark-grey: #474c55;
    --green: #6db420;
    --yellow: #fec01e;
}

body {
  background-color: var(--off-white);
  margin:0;
  padding:0;
  font-family: 'Open Sans',  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html {
  box-sizing: border-box;
}
*, *:before, *:after {
  box-sizing: inherit;
}

.page-header {
  font-size: 34px;
  margin: 50px 0;
}
.page-paragraph {
  font-size: 24px;
  margin: 10px 0 30px 0;
  color: var(--dark-grey)
}

.display {
  &--flex {
    display: flex;
  };
  &--block {
    display: block;
  }
  &--inline-block {
    display: inline-block;
  }
}

.space-between {
  justify-content: space-between;
}

.align-self {
  &--center {
    align-self: center;
  }
  &--flex-end{
    align-self: flex-end;
  }
  &--flex-start{
    align-self: flex-start;
  }
}

.container {
  position: relative;
  max-width: 1100px;
  margin: 0 auto;
}

a {
  text-decoration: none;
  color: var(--dark-grey);
  &:hover {
    color: black;
  }
}

.text {
  &-weight {
    &--bold {
      font-weight: bold;
    }
  }
  &-align {
    &--center {
      text-align: center;
    }
  }
  &--decoration-none {
    text-decoration: none
  }
  &--indent {
    text-indent: -10000000px;
  }
  &-transform {
    &--uppercase {
      text-transform: uppercase;
    }
    &--capitalize {
      text-transform: capitalize;
    }
  }
}

input.error {
  box-shadow: 0 0 20px var(--yellow);
  background: var(--yellow);
}


