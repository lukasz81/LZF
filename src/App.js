import React, { Component } from 'react';
import Header from './header/Header';
import Widget from './widget/Widget';
import Content from './content/Content';
import Footer from './footer/Footer';
import './App.css';

export class App extends Component {
  render() {
    return (
      <div className="App">
          <Header/>
          <Widget/>
          <Content/>
          <Footer/>
      </div>
    );
  }
}

export default App;
