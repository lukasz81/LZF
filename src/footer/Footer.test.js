import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ConnectedFooter,{Footer} from './Footer';
import App from '../App';

Enzyme.configure({ adapter: new Adapter() });

describe('<Footer/>', () => {

    it('renders Footer component in App', () => {
        const container = shallow(<App/>);
        expect(container.find(ConnectedFooter).length).toEqual(1);
    });

    it('renders Footer snapshot correctly', () => {
        const content = TestRenderer
            .create(<Footer/>)
            .toJSON();
        expect(content).toMatchSnapshot();
    });

});