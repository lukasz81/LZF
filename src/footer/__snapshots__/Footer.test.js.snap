// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders Footer snapshot correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="container display--flex"
  >
    <ul>
      <li>
        <h4
          className="text-transform--uppercase"
        >
          Who we are
        </h4>
      </li>
      <li>
        <a
          href="./"
        >
          Our history
        </a>
      </li>
      <li>
        <a
          href="./"
        >
          Press center
        </a>
      </li>
      <li>
        <a
          href="./"
        >
          Contact us
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <h4
          className="text-transform--uppercase"
        >
          Get involved
        </h4>
      </li>
      <li>
        <a
          href="./"
        >
          Join us
        </a>
      </li>
      <li>
        <a
          href="./"
        >
          Donate
        </a>
      </li>
      <li>
        <a
          href="./"
        >
          Careers
        </a>
      </li>
      <li>
        <a
          href="./"
        >
          Shop
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <h4
          className="text-transform--uppercase"
        >
          Follow us
        </h4>
      </li>
      <div
        className="display--flex social"
      >
        <li>
          <a
            href="./"
          >
            <img
              alt="Facebook"
              src="./images/facebook-logo.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="./"
          >
            <img
              alt="Pinterest"
              src="./images/pinterest-logo.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="./"
          >
            <img
              alt="Twitter"
              src="./images/twitter-logo.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="./"
          >
            <img
              alt="Youtube"
              src="./images/youtube-logo.svg"
            />
          </a>
        </li>
      </div>
    </ul>
    <ul>
      <h4>
        <img
          alt="logo"
          src="./images/lts-logo-light.svg"
        />
      </h4>
      <li>
        3 Warren Rd,
        <br />
        Mary Tavy,
        <br />
        Tavystock PL19 9PF,
        <br />
        United Kingdom
      </li>
    </ul>
  </div>
</footer>
`;
